superset/db_engine_specs/base.py

```diff
@@ -1202,6 +1206,74 @@ def convert_dttm(  # pylint: disable=unused-argument
         """
         return None
 
+    @classmethod
+    def handle_boolean_filter(
+        cls, sqla_col: Any, op: str, value: bool
+    ) -> BinaryExpression:
+        """
+        Handle boolean filter operations with engine-specific logic.
+
+        By default, uses SQLAlchemy's IS operator (column IS true/false).
+        Engines that don't support IS for boolean values can override
+        use_equality_for_boolean_filters to use equality operators instead.
+
+        :param sqla_col: SQLAlchemy column element
+        :param op: Filter operator (IS_TRUE or IS_FALSE)
+        :param value: Boolean value (True or False)
+        :return: SQLAlchemy expression for the boolean filter
+        """
+        if cls.use_equality_for_boolean_filters:
+            return sqla_col == value
+        else:
+            return sqla_col.is_(value)
+
+    @classmethod
+    def handle_null_filter(cls, sqla_col: Any, op: str) -> BinaryExpression:
```

Comment:
We could be more strict in the `op` type here, to have the type checker catch incorrect calls to this method:

```suggestion
    def handle_null_filter(
        cls,
        sqla_col: Any,
        op: utils.FilterOperator.IS_NULL | utils.FilterOperator.IS_NOT_NULL,
    ) -> BinaryExpression:
```